src/System/Hapistrano/Types.hs

```diff
@@ -57,15 +60,28 @@ data Config =
     -- ^ How to print messages
     }
 
+-- | The origin of the repository. It can be from a version control provider
+-- like GitHub or a local directory.
+data Repository
+  = ExternalRepository
```

Comment:
Since we could add support for other sources, this constructor and its fields could be better named after Git ...

Reply:
I think we could take this in a separate PR, since that involves other changes.

Reply:
Since this PR is introducing these new constructors, then it could be modified now. On second thought, `GitRepository` might be a better name for this constructor and a similar approach could be followed for the fields on the record